### PR TITLE
feat(service overview): add deactivate

### PR DESCRIPTION
**Pull request by lavanya-bmw**
## Description

Added deactivate functionality to service overview for active services

## Why

To enhance active services

## Issue

Ref: CPLP-2821

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

File: src/components/pages/ServiceReleaseProcess/components/ServiceDeactivate.tsx [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    }
+  }, [fetchImage, leadImageId])
+
+  const handleSaveClick = async () => {
+    setIsLoading(true)
+    await deactivateService(service[0].id)
+      .unwrap()
+      .then(() =>
+        navigate('/serviceoverview', {
+          state: 'service-deactivate-success',
```
Comment (manojava-gk) [inline src/components/pages/ServiceReleaseProcess/components/ServiceDeactivate.tsx:87]: create an enum type

File: src/components/pages/ServiceReleaseProcess/components/ServiceDeactivate.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    async (documentId: string, docType: string) => {
+      try {
+        const response = await fetchDocumentById({
+          appId: serviceId ?? '',
+          documentId,
+        })
+
+        if (response && 'data' in response) {
+          const file = response?.data?.data
+          if (docType === 'APP_LEADIMAGE') {
```
Comment (manojava-gk) [inline src/components/pages/ServiceReleaseProcess/components/ServiceDeactivate.tsx:64]: use enum type

**Review by manojava-gk — COMMENTED** [review]
small finding related to data types

**Review by manojava-gk — APPROVED** [review]
lftm

File: src/components/pages/ServiceReleaseProcess/components/ServiceDeactivate.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+      }
+    },
+    [fetchDocumentById, serviceId]
+  )
+
+  useEffect(() => {
+    if (leadImageId) {
+      fetchImage(leadImageId, DocumentTypeId.APP_LEADIMAGE)
+    }
+  }, [fetchImage, leadImageId])
```
Comment (oyo) [inline src/components/pages/ServiceReleaseProcess/components/ServiceDeactivate.tsx:82]: Don't write custom fetch image code - use shared <Image> component and pass url along with ´ImageService.fetchImageWithToken´.

**Review by oyo — CHANGES_REQUESTED** [review]
redundant fetchImage code

File: src/components/pages/ServiceReleaseProcess/components/ServiceDeactivate.tsx (hunk 3 above)
Comment (lavanya-bmw) [reply src/components/pages/ServiceReleaseProcess/components/ServiceDeactivate.tsx]: @oyo Addressed comments

File: src/assets/locales/de/servicerelease.json [hunk 4]
```diff
@@ -16,6 +16,13 @@
   "loadmore": "Mehr laden",
   "documentDeleteSuccess": "Document deleted successfully",
   "documentDeleteError": "Something went wrong.",
+  "global": {
+    "actions": {
+      "confirm": "Confirm",
+      "cancel": "Cancel",
+      "save": "Save"
+    }
+  },
```
Comment (jjeroch) [inline src/assets/locales/de/servicerelease.json:25]: @lavanya to reduce future efforts can you please directly add the german translation. As far as i remember Martin had a tool for that - isnt it? @oyo

File: src/assets/locales/de/servicerelease.json [hunk 5]
```diff
@@ -120,6 +127,14 @@
...
     "sortOptions": {
       "deactivate": "Deactivate"
+    },
+    "serviceDeactivate": {
+      "headerTitle": "Deactivate Service",
+      "description": "Deactivation of service will deactivate the service offer inside the Catena-X marketplace. Any current active subscription will stay active and customers wont be effected by the deactivation. New subscriptions are not possible.",
+      "checkboxLabel": "Hereby I confirm that I have read the business service terms and conditions and that I am deactivating my company business services from the marketplace with immediate effect. I understand that it is not possible to self-reverse the deactivation.",
+      "checkboxErrorMsg": "Please confirm that you have read the deactivation impact above before proceeding",
+      "successMsg": "Deactivation successfully completed",
+      "errorMsg": "Error! Something went wrong"
```
Comment (jjeroch) [inline src/assets/locales/de/servicerelease.json:137]: same here - translation needed to reduce future efforts

**Review by jjeroch — CHANGES_REQUESTED** [review]

File: src/assets/locales/de/servicerelease.json (hunk 5 above)
Comment (lavanya-bmw) [reply src/assets/locales/de/servicerelease.json]: Translated using https://translate.google.com/

**Comment by jjeroch** [comment]
> redundant fetchImage code

@lavanya-bmw is this done - if so - please assign (via the top right menu) the pr to martin (via the assignees tag)

File: src/components/pages/ServiceReleaseProcess/components/ServiceDeactivate.tsx [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        .then((buffer) =>
+          setDeactivateCardImage(
+            URL.createObjectURL(new Blob([buffer], { type: 'image/png' }))
+          )
+        )
+        .catch((err) => {
+          error('ERROR WHILE FETCHING IMAGE', '', err as object)
+        })
+    }
+  }, [leadImageId])
```
Comment (oyo) [inline src/components/pages/ServiceReleaseProcess/components/ServiceDeactivate.tsx:70]: don't load image by yourself - instead pass the image url directly to the Card and 
`imageLoader={fetchImageWithToken}`

File: src/components/pages/AppOverview/Deactivate.tsx [hunk 7]
```diff
@@ -47,34 +48,23 @@ export default function Deactivate() {
...
+          setDeactivateCardImage(
+            URL.createObjectURL(new Blob([buffer], { type: 'image/png' }))
+          )
+        )
+        .catch((err) => {
+          error('ERROR WHILE FETCHING IMAGE', '', err as object)
+        })
     }
-  }, [fetchImage, leadImageId])
+  }, [leadImageId])
```
Comment (oyo) [inline src/components/pages/AppOverview/Deactivate.tsx:67]: don't load image by yourself - instead pass the image url directly to the Card and 
`imageLoader={fetchImageWithToken}`

**Review by oyo — CHANGES_REQUESTED** [review]

**Review by oyo — APPROVED** [review]

**Pull request merged**